### PR TITLE
framework/st_things : st things data manager refactoring

### PR DESCRIPTION
**Pull request by JeongJunSik**
- Remove unused variables
- Add "speckVersion", "dataModelVersion" JSON parsing logic
- Change the return logic due to JSON error
- Remove init_things_info_file function

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:70: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/st_things/things_stack/framework/things_data_manager.c:71: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/st_things/things_stack/framework/things_data_manager.c:72: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/st_things/things_stack/framework/things_data_manager.c:73: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/st_things/things_stack/framework/things_data_manager.c:74: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/st_things/things_stack/framework/things_data_manager.c:126: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 1]
```diff
@@ -43,114 +43,101 @@
...
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_MANUFACTURERURL		"manufacturerUrl"
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_MANUFACTURINGDATE		"manufacturingDate"
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_MODELNUMBER			"modelNumber"
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_PLATFORMVERSION		"platformVersion"
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_OSVERSION				"osVersion"
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_HARDWAREVERSION		"hardwareVersion"
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_FIRMWAREVERSION		"firmwareVersion"
+#define KEY_DEVICE_SPECIFICATION_PLATFORM_VENDORID				"vendorId"
+
+/* resources */
```
Comment (juitem) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: it would be better align spaces

Comment (JeongJunSik) [reply framework/src/st_things/things_stack/framework/things_data_manager.c]: OK, i fixed

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**